hs/WebParsing/TimeTableParser.hs

```diff
@@ -17,15 +18,84 @@ import WebParsing.TimeConverter
 import Control.Monad.IO.Class (liftIO, MonadIO)
 import Control.Monad.Trans.Reader (ReaderT)
 import Config (databasePath, timetableUrl)
+import Database.Tables (Department(Department))
 
 parseTT :: IO ()
 parseTT = do
     rsp <- simpleHTTP (getRequest timetableUrl)
     body <- getResponseBody rsp
+
     let depts = getDeptList $ parseTags body
+        deptCode = getDeptCodes $ parseTags body
+        deptName = getDeptName $ parseTags body
+        deptAssocList = init $ constructAssociationList deptName deptCode
 
     runSqlite databasePath $
         mapM_ getDeptTimetable depts
+    runSqlite databasePath $ 
+        mapM_ insertDepartment deptAssocList
+
+-- | List of unwanted words to remove when getting department codes.
+wordsToRemove :: [String]
+wordsToRemove = ["courses",
+                "courses]", 
+                "includes",
+                "and",
+                "-",
+                "[",
+                "]",
+                ""]
+
+-- | Extracts a list of lists of department codes from the main website
+getDeptCodes :: [Tag String] -> [[T.Text]]
+getDeptCodes tags = 
+    let tagsText = map fromTagText $ filter isTagText tags
+        -- filterDeptInfo returns only the strings that contain dept codes
+        filterDeptInfo = map (charInString '[') tagsText
```

Comment:
These two lines (and the helper function below) are too complex.

Using filter is good; make use of the `elem` function rather than defining `charInString`.